dshr/dshr_mod.F90

```diff
@@ -320,39 +332,80 @@ subroutine dshr_mesh_init(gcomp, compid, logunit, compname, model_nxg, model_nyg
        ! TODO: need a check that the mask file has the same grid as the model mesh
        ! Reset the model mesh mask if the mask file is different from the mesh file
        if (trim(model_meshfile) /= trim(model_maskfile)) then
-          pio_subsystem => shr_pio_getiosys(compid)
-          io_type       =  shr_pio_getiotype(compid)
-          io_format     =  shr_pio_getioformat(compid)
+
+          pio_subsystem => shr_pio_getiosys(trim(compname))
+          io_type       =  shr_pio_getiotype(trim(compname))
+          io_format     =  shr_pio_getioformat(trim(compname))
 
           ! obtain lsize and model_gindex
           call ESMF_MeshGet(model_mesh, elementdistGrid=distGrid, rc=rc)
           if (ChkErr(rc,__LINE__,u_FILE_u)) return
           call ESMF_DistGridGet(distGrid, localDe=0, elementCount=lsize, rc=rc)
           if (ChkErr(rc,__LINE__,u_FILE_u)) return
+
+          ! allocate memory for model_mask and model_frac
+          allocate(model_frac(lsize))
+          allocate(model_mask(lsize))
+
+          ! get model_gindex (allocate memory first)
           allocate(model_gindex(lsize))
-          allocate(elemMask(lsize))
           call ESMF_DistGridGet(distGrid, localDe=0, seqIndexList=model_gindex, rc=rc)
           if (ChkErr(rc,__LINE__,u_FILE_u)) return
 
-          ! obtain model mask from separate model_maskfile (assume mask name on mask file)
+          ! obtain model mask and model frac from separate model_maskfile
           rcode = pio_openfile(pio_subsystem, pioid, io_type, trim(model_maskfile), pio_nowrite)
           call pio_seterrorhandling(pioid, PIO_BCAST_ERROR)
```

Comment:
This should be pio_seterrorhandling(pioid, PIO_BCAST_ERROR, old_error_handle)
then in line 362 pio_seterrorhandling(pioid, old_error_handle)
Same with lines 369 and 373

Reply:
Can you please clarify what you want here - where are you defining old_error_handle?

Reply:
Old_error_handle is an output in the first call and is set to whatever the value was previous to the change.